app/misc.py

```diff
@@ -1167,6 +1167,50 @@ def getSubData(sid, simple=False, extra=False):
     return data
 
 
+def getModSubs(uid):
```

Comment:
This function could accept a `power_level` parameter and then just query for `SubMod.power_level <= power_level`. It would make it simpler and more reusable in the future

Reply:
I actually legit have no idea how this even works - I just shamelessly copied the code from elsewhere :sweat_smile:  Can you spell it out for me a little more? I don't really get how the powerlevel works or where that comes from

Reply:
We use the `power_level` field in the mods table to store *what kind of mod* a user is. We have the following power levels:

 - 0: Owner. Can do everything
 - 1: Moderators: Can do everything except adding or removing mods
 - 2: Janitors: Can only delete posts/comments, ban users (only timed bans, not permanent ones), flair posts and everything that is not editing the sub's settings

This function could be simplified by taking a power level number and only returning the users that have that power level or less. You could use the `is_sub_mod` function as a reference if you want to see the peewee syntax. This could save you the owns/mods/subs lines

Reply:
Ahh okay, thanks! I will look into that :)

Reply:
Okay, I think the change I just pushed does this!